internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -836,6 +842,35 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	return resourceRead(ctx, d, meta)
 }
 
+func isUpdateAllowed(d *schema.ResourceData) (bool, error) {
+	cs, us := d.GetChange("replication_specs")
+	currentSpecs, updatedSpecs := cs.([]any), us.([]any)
+
+	isNewSchemaCompatible := checkNewSchemaCompatibility(currentSpecs)
+
+	for _, specRaw := range updatedSpecs {
+		if specMap, ok := specRaw.(map[string]any); ok && specMap != nil {
+			numShards, _ := specMap["num_shards"].(int)
+			if numShards > 1 && isNewSchemaCompatible {
+				return false, fmt.Errorf("cannot increase num_shards to >1 under the current configuration. Please modify your resource as outlined in our migration guide before updating to add additional shards")
+			}
+		}
+	}
+	return true, nil
+}
+
+func checkNewSchemaCompatibility(specs []any) bool {
+	for _, specRaw := range specs {
+		if specMap, ok := specRaw.(map[string]any); ok && specMap != nil {
+			numShards, _ := specMap["num_shards"].(int)
+			if numShards < 2 {
+				return true
+			}
+		}
+	}
+	return false
```

Comment:
Would consider that for geosharded cluster you can have a first replication spec with numShards=1, and a second with numShards=2. For this case I believe the current logic would fail.

Reply:
you're right, updated the logic to check for all replication spec numShards